sound/soc/sof/topology.c

```diff
@@ -2163,6 +2163,11 @@ void snd_sof_free_topology(struct snd_sof_dev *sdev)
 		snd_soc_dapm_del_routes(dapm, sroute->route, 1);
 
 		/* free sroute and its private data */
+		kfree(sroute->route->source);
+		kfree(sroute->route->sink);
+		if (sroute->route->control)
```

Comment:
no need to check if (ptr) here as kfree() does this already.

Reply:
@lgirdwood 
ok, I will update it after back to office.
thanks